src/Fixer/FunctionNotation/CombineNestedDirnameFixer.php

```diff
@@ -181,25 +181,43 @@ private function getDirnameInfo(Tokens $tokens, $index, $firstArgumentEndIndex =
         }
 
         $next = $tokens->getNextMeaningfulToken($next);
+        $info['indexes'][] = $next;
+
+        if ($tokens[$next]->equals(')')) {
+            $info['levels'] = 1;
+            $info['end'] = $next;
+
+            return $info;
+        }
 
         if (!$tokens[$next]->isGivenKind(T_LNUMBER)) {
             return false;
         }
 
-        $info['indexes'][] = $next;
         $info['secondArgument'] = $next;
         $info['levels'] = (int) $tokens[$next]->getContent();
 
         $next = $tokens->getNextMeaningfulToken($next);
 
-        if (!$tokens[$next]->equals(')')) {
-            return false;
+        if ($tokens[$next]->equals(')')) {
```

Comment:
maybe move this after the `equals(',')` check?

```php
	        if ($tokens[$next]->equals(',')) {
	            $info['indexes'][] = $next;
	            $next = $tokens->getNextMeaningfulToken($next);
	        }

	        if (!$tokens[$next]->equals(')')) {
	            return false;
	        }

	        $info['indexes'][] = $next;
	        $info['end'] = $next;
	
	        return $info;
```

Reply:
Cool, thanks for both comments - looks much better now.